src/sprites/Elevator.lua

```diff
@@ -72,118 +66,225 @@ function Elevator:collisionResponse(_)
   return gfx.sprite.kCollisionTypeSlide
 end
 
---- Used specifically for when jumping while moving up with elevator.
-function Elevator:disableCollisionsForFrame()
-  self:setCollisionsEnabled(false)
+--
+--
+----- Private class methods
+--
 
-  self.isCollisionsDisabledForFrame = true
-end
+local function getActivationMovement(self, key)
+  if self.fields.orientation == ORIENTATION.Horizontal then
+    -- Horizontal orientation, return positive if Right, negative if Left
 
--- Private class methods
+    if key == KEYNAMES.Right then
+      return self.speed
+    elseif key == KEYNAMES.Left then
+      return -self.speed
+    end
+  else
+    -- Vertical orientation, return positive if Down, negative if Up
 
---- Get remaining movement based on direction and displacement
-local function getMovementRemaining(self, key)
-  -- Get inverted direction boolean
-  local isInverseDirection = isInverseDirection(key)
-
-  -- Remaining displacement - either towards displacementEnd or displacementStart if inverse direction
-  local displacementRemaining = isInverseDirection 
-    and self.displacement - self.displacementStart
-    or self.displacementEnd - self.displacement
-  
-  -- Calculate movement as scalar value, invert speed if inverse direction
+    if key == KEYNAMES.Down then
+      return self.speed
+    elseif key == KEYNAMES.Up then
+      return -self.speed
+    end
+  end
 
-  return math.min(displacementRemaining, self.speed) * (isInverseDirection and -1 or 1)
-end
+  -- If key is not a correct activation key, return 0
```

Comment:
should we consider asserting more strongly? (there are a few fallbacks like this, same question for all of them)

Reply:
Need to double-check that Player isn't pushing random other keys to the elevator. I wouldn't have considered it an error necessarily that the elevator receives other keys but for this version you have a good point.

Reply:
Player is checking random keys, so added the assert :)